### PR TITLE
Add listing to stow storage apis

### PR DESCRIPTION
**Pull request by bgedik**
## Tracking issue
Part of RFC https://github.com/flyteorg/flyte/blob/b3add894fd879d6b6c11e37ba292ae04bf60afeb/rfc/system/5598-deterministic-errors-distributed-training.md#flytepropeller-backend

## Why are the changes needed?

Deterministic error propagation, see: https://github.com/flyteorg/flyte/blob/b3add894fd879d6b6c11e37ba292ae04bf60afeb/rfc/system/5598-deterministic-errors-distributed-training.md#flytepropeller-backend

## What changes were proposed in this pull request?

Adding listing APIs

## How was this patch tested?

Unit tests

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [X] All commits are signed-off.

## Related PRs

https://github.com/flyteorg/flyte/pull/5616

## Docs link

N/A

### CODE REVIEW

File: flyteplugins/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go
Comment (bgedik) [inline flyteplugins/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go:63]: ignore

File: flyteplugins/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds_test.go
Comment (bgedik) [inline flyteplugins/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds_test.go:27]: ignore

File: flyteplugins/go/tasks/plugins/k8s/spark/spark_test.go
Comment (bgedik) [inline flyteplugins/go/tasks/plugins/k8s/spark/spark_test.go:853]: ignore

File: flytestdlib/storage/storage.go
Comment (wild-endeavor) [inline flytestdlib/storage/storage.go:46]: Could we add some comments to these numbers and to the struct please?  Also can we rename the constants?  These aren't really cursor positions right?  These are more like cursor types.

Comment (bgedik) [reply flytestdlib/storage/storage.go:46]: will do

**Review by wild-endeavor — APPROVED** [review]
thanks but let me get one more approval on this before merging as well.

**Review by EngHabu — DISMISSED** [review]

**Comment by kumare3** [comment]
If we support listing we can do map over directories

**Comment by bgedik** [comment]
Messed up the DCO pretty bad, will create a new one

**Comment by bgedik** [comment]
https://github.com/flyteorg/flyte/pull/5741

**Pull request closed without merging**
